OSDOCS-11208-installer: Bug batched 4.17 Installer bugs (#82484)
openshift/openshift-docs: release_notes/ocp-4-17-release-notes.adoc

* Previously, when installing a cluster on {aws-short}, subnets that the installation program created were incorrectly tagged with the "kubernetes.io/cluster/<clusterID>: shared" tag. With this update, these subnets are correctly tagged with the "kubernetes.io/cluster/<clusterID>: owned" tag. (link:https://issues.redhat.com/browse/OCPBUGS-36904[*OCPBUGS-36904*])

* Previously, the local etcd data store that is saved during installation was not deleted if the installation failed, consuming extra space on the installation host. With this update, the data store is deleted if infrastructure provisioning failures prevent a successful installation. (link:https://issues.redhat.com/browse/OCPBUGS-36284[*OCPBUGS-36284*])

> * Previously, when a folder was undefined and the data center was located in a datacenter folder, an wrong folder structure was created starting from the root of the vCenter server. By using the Govmomi `DatacenterFolders.VmFolder`, it used the an wrong path. With this release, the folder structure uses the data center inventory path and joins it with the virtual machine (VM) and cluster ID value, and the issue is resolved.(link:https://issues.redhat.com/browse/OCPBUGS-38616[*OCPBUGS-38616*])

Reviewer: ```suggestion
* Previously, when a folder was undefined and the data center was located in a data center folder, a wrong folder structure was created starting from the root of the vCenter server. By using the Govmomi `DatacenterFolders.VmFolder`, it used the an wrong path. With this release, the folder structure uses the data center inventory path and joins it with the virtual machine (VM) and cluster ID value, and the issue is resolved.(link:https://issues.redhat.com/browse/OCPBUGS-38616[*OCPBUGS-38616*])
```
See ISG: https://www.ibm.com/docs/en/ibm-style?topic=word-usage#d

Author: :heavy_check_mark:

* Previously, due to an EFI Secure Boot failure in the SCOS, when the FCOS pivoted to the SCOS the virtual machine (VM) failed to boot. With this release, the Secure Boot is disabled only when the Secure Boot is enabled in the `coreos.ovf ` configuration file, and the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-37736[*OCPBUGS-37736*])

* Previously, if you specified an unsupported architecture in the `install-config.yaml` file the installation program would fail with a "connection refused" message. With this update, the installation program correctly validates the cluster architecture parameter, leading to successful installations. (link:https://issues.redhat.com/browse/OCPBUGS-38841[*OCPBUGS-38841*])

* Previously, a rare condition om {vmw-full} Cluster API machines caused the vCenter session management to time out unexpectedly. With this release, the Keep Alive support is disabled in the current and later versions of CAPV, and the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-38677[*OCPBUGS-38677*])

> * Previously, the installation program on {aws-first} used multiple IPv4 public IPs that Amazon has started charging for. With this release, support is provided for bring your own (BYO) public IPv4 pools in {product-title} so that users have control of IPs that are used by their services. Where the BYO public IPv4 pools feature is enabled, two new permissions, `ec2:DescribePublicIpv4Pools` and `ec2:DisassociateAddress`, are required, and the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-35504[*OCPBUGS-35504*])

Reviewer: ```suggestion
* Previously, the installation program on {aws-first} used multiple IPv4 public IP addresses that Amazon has started charging for. With this release, support is provided for bring your own (BYO) public IPv4 pools in {product-title} so that users have control of IP addresses that are used by their services. Where the BYO public IPv4 pools feature is enabled, two new permissions, `ec2:DescribePublicIpv4Pools` and `ec2:DisassociateAddress`, are required, and the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-35504[*OCPBUGS-35504*])
```
See ISG: https://www.ibm.com/docs/en/ibm-style?topic=word-usage#i

Author: :heavy_check_mark:

* Previously, when installing a cluster with the Agent-based installer, the assisted-installer process could timeout when attempting to add control plane nodes to the cluster. With this update, the assisted-installer process loads fresh data from the assisted-service process, preventing the timeout. (link:https://issues.redhat.com/browse/OCPBUGS-36779[*OCPBUGS-36779*])

* Previously, when the {vmw-full} vCenter cluster contained an ESXi host that did not have a standard port group defined and the installation program tried to select that host to import the OVA, the import failed and the error “Invalid Configuration for device '0'” was presented.
With this release, the installation program verifies whether a standard port group for an ESXi host is defined and, if not, continues until it locates an ESXi host with a defined standard port group or presents an error message if it fails to locate one, resolving the issue. 
 (link:https://issues.redhat.com/browse/OCPBUGS-38560[*OCPBUGS-38560*])

> * Previously, extracting the IP address from the Cluster API Machine object only returned a single address. On {vmw-first}, the returned address would always be an IPv6 address and this caused issues with the `must-gather` implementation if the address was non-routable. With this release, the Cluster API Machine object returns all IP addresses, including IPv4, so that the `must-gather` issue no longer occurs on {vmw-full}. (link:https://issues.redhat.com/browse/OCPBUGS-37607[*OCPBUGS-37607*])

Reviewer: ```suggestion
* Previously, extracting the IP address from the Cluster API Machine object only returned a single IP address. On {vmw-first}, the returned address would always be an IPv6 address and this caused issues with the `must-gather` implementation if the address was non-routable. With this release, the Cluster API Machine object returns all IP addresses, including IPv4, so the `must-gather` issue no longer occurs on {vmw-full}. (link:https://issues.redhat.com/browse/OCPBUGS-37607[*OCPBUGS-37607*])
```
See ISG: https://www.ibm.com/docs/en/ibm-style?topic=word-usage#i

Author: :heavy_check_mark:
